File: internal/core/ports/storage.go
package ports

type Storage interface {
	SaveDriver(name string, chatID int64) error
	GetDrivers(chatID int64) ([]string, error)
	IsDriverExists(name string, chatID int64) (bool, error)
	SavePassenger(name string, chatID int64) error
	GetPassengers(chatID int64) ([]string, error)
	SaveDeparture(driverName string, passengers []string, chatID int64) error
	SaveReturn(driverName string, passengers []string, chatID int64) error
	GetPassengerTripPrice(passengerName string, chatID int64) (int, error)
	GetDeparturePassengers(driverName string, chatID int64) ([]string, error)
	HasDepartureToday(passengerName string, chatID int64) (bool, error)
	GetTodayDrivers(chatID int64) ([]string, error)
	GetReturnPassengers(driverName string, chatID int64) ([]string, error)
}

File: src/core/ports/storage.ts
export interface Storage {
    saveDriver(name: string, chatId: number): Promise<void>;
    getDrivers(chatId: number): Promise<string[]>;
    isDriverExists(name: string, chatId: number): Promise<boolean>;
    savePassenger(name: string, chatId: number): Promise<void>;
    getPassengers(chatId: number): Promise<string[]>;
    saveDeparture(driverName: string, passengers: string[], chatId: number): Promise<void>;
    saveReturn(driverName: string, passengers: string[], chatId: number): Promise<void>;
    getPassengerTripPrice(passengerName: string, chatId: number): Promise<number>;
    getDeparturePassengers(driverName: string, chatId: number): Promise<string[]>;
    hasDepartureToday(passengerName: string, chatId: number): Promise<boolean>;
    getTodayDrivers(chatId: number): Promise<string[]>;
    getReturnPassengers(driverName: string, chatId: number): Promise<string[]>;
    getDriversByDate(chatId: number, date: string): Promise<string[]>;
    getDeparturePassengersByDate(driverName: string, chatId: number, date: string): Promise<string[]>;
    getReturnPassengersByDate(driverName: string, chatId: number, date: string): Promise<string[]>;
}